### PR TITLE
test: check for status codes that are different on oCIS and reva

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
These test scenarios in `webDavSpecialURLs.feature` were recently updated to specify the actual HTTP status codes that are returned by oCIS. But when we run these scenarios on reva we get slightly different HTTP status codes.

These are tests for unusual edge-case calls to end-points, using double-slash in unexpected parts of the URL. The main thing is that these unusual cases do not "crash" the server, and they do return "some reasonable response". If oCIS and reva are a little different then it does not really matter.

A new step definition has been added that allows us to specify the HTTP status codes expected on oCIS and on reva as separate/different lists. That step definition is used in the tests to specify what we observe as the current behavior on oCIS and reva.

This problem was noticed when trying to bump the oCIS commit id in reva PR https://github.com/cs3org/reva/pull/4715

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiAuth/webDavSpecialURLs.feature [hunk 1]
```diff
@@ -20,7 +20,7 @@ Feature: make webdav request with special urls
       | /remote.php//dav/files/%username%/PARENT/parent.txt |
       | /remote.php//webdav/PARENT                          |
       | //remote.php/dav//files/%username%//FOLDER          |
-    Then the HTTP status code of responses on each endpoint should be "200,200,204,204,200" respectively
+    Then the HTTP status code of responses on each endpoint should be "200,200,204,204,200" on oCIS and "204,204,204,204,204" on reva
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiAuth/webDavSpecialURLs.feature:23]: should we use `or` here because only one of the two will be true at a given test context
```suggestion
    Then the HTTP status code of responses on each endpoint should be "200,200,204,204,200" on oCIS or "204,204,204,204,204" on reva
```

Comment (phil-davis) [reply tests/acceptance/features/coreApiAuth/webDavSpecialURLs.feature]: good suggestion - done

**Review by saw-jan — APPROVED** [review]
